app/controllers/works_controller.rb

```diff
@@ -2,6 +2,7 @@ class WorksController < ApplicationController
   # We should always be able to tell what category
   # of work we're dealing with
   before_action :category_from_work, except: [:root, :index, :new, :create]
+  skip_before_action :find_user, only: [:root]
```

Comment:
By avoiding this filter it shows a log in button on the homepage, even after I'm logged in.  You can add a different bit of ocood to the root path to find the user.  Or create 2 filters, one to find the user and another to redirect if it's not there.